solr/core/src/java/org/apache/solr/servlet/PriorityBasedRateLimiter.java

```diff
@@ -111,7 +114,7 @@ private void exitFromQueue() {
 
   private void release() {
     this.exitFromQueue();
-    if (this.activeRequests.get() < this.totalAllowedRequests) {
+    if (this.activeRequests.get() < this.backgroundRequestAdmitThreshold) {
       // next priority
       CountDownLatch waiter = this.waitingList.poll();
       if (waiter != null) {
```

Comment:
wait, what happens if there's a FOREGROUND request waiting on `waitingList`? Don't we want to wake up waiting fg requests once active requests drops below `totalAllowedRequests`?

Reply:
✔️ right, I see 🤦